### ISSUE TITLE
Raspberry Pi 3

### ISSUE THREAD

**Issue by henryp007** [OpenLightingProject/ola#1212]
Hi wondering if you can help? just tried running the OLA image on my RPI 3 and i only get the default multi colour boot screen and nothing else. does this software work on the RPI 3 yet or am i doing something wrong?

i have run other images on this device so i know it's not a hardware issue.
thanks !

**Comment by peternewman** [comment]
See https://github.com/OpenLightingProject/raspberrypi/issues/7 for tracking of this issue @henryp007 . Essentially the "current" Pi images don't support the Pi 3 hardware.

However OLA is in Raspbian Jessie, so just install that and then "sudo apt-get install ola" should do the trick.

**Comment by henryp007** [comment]
Thanks! Will this work the same as the Pi image would? 
Do you know if there's any plans to release an updated image for the Pi 3?

**Comment by atom-42** [comment]
I'm working on that. I have very few free time these days, but I have a working rpi3 image. It needs to be cleaned and a proper build script before it can be released. Apart from not being clean, it works fine.

**Comment by henryp007** [comment]
Awesome 👍 I look forward to it being released!

**Comment by peternewman** [comment]
Hi @henryp007,

Yes, installing the debs would get you the same as the deb version of the Pi image. The images are just designed to make it easier for people to get started.

**Comment by atom-42** [comment]
It's been something like a billion years... But I finally found enough time to write something a bit more user friendly.
You can find that here : https://github.com/atom-42/ola-rpi3-image-builder
I can provide images made with the script too.

**Comment by peternewman** [comment]
Thanks for the efforts @atom-42 , however similar to previous attempts you should really look at doing your modifications to config.txt and other system files as patches, in case Raspberry Pi add more options in future:
https://github.com/OpenLightingProject/raspberrypi/pull/1#issuecomment-61378383

Also you probably want to use our https://github.com/OpenLightingProject/raspberrypi/tree/master/etc and https://github.com/OpenLightingProject/raspberrypi/tree/master/home/pi directories via some git submodules, so you track any further changes we make in them in future.

**Comment by atom-42** [comment]
Thank you for the feedback @peternewman , i'll look into that as soon as i can.

**Comment by ylevtov** [comment]
Hi @peternewman - is it possible that the ola version in Raspbian Jessie doesn't work on the Pi3 because of a couple of other dependencies? Currently I'm stuck on trying to install `libprotobuf7`, and consequently `libola1`. Just checking I'm not barking up the wrong tree! 

I'd love to install ola onto my current build of Raspbian rather than installing via an image as I've got a lot of other content already on the Pi I'd like to keep...

**Comment by peternewman** [comment]
Hi @ylevtov ,

If you've got Jessie, you should just be able to apt-get install ola, you don't need the OLA APT repo, or to install any other packages, it will pull in the dependencies it needs.

**Comment by ylevtov** [comment]
@peternewman 
Hmm, I tried that a few times but get the following:
```
pi@raspberrypi:~ $ sudo apt-get install ola
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 ola : Depends: libola1 (>= 0.9.8) but it is not going to be installed
       Depends: libprotobuf7 but it is not installable
E: Unable to correct problems, you have held broken packages.
```

Double checking I'm actually on Jessie:
```
pi@raspberrypi:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 8 (jessie)"
NAME="Raspbian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

Trying to install `libprotobuf7` manually also fails, (but installing `libprotobuf9` works). Is it possible to point `libola1` at `libprotobuf9` rather than `libprotobuf7`?

Thanks for any help!

**Comment by peternewman** [comment]
What does `apt-cache policy ola` show? I'm going to guess that lists an Open Lighting URL too?

**Comment by ylevtov** [comment]
Here's the output:
```
pi@raspberrypi:~ $ apt-cache policy ola
ola:
  Installed: (none)
  Candidate: 0.9.8-1~wheezy1
  Version table:
     0.9.8-1~wheezy1 0
        500 http://apt.openlighting.org/raspbian/ wheezy/main armhf Packages
     0.9.1-1.1 0
        500 http://mirrordirector.raspbian.org/raspbian/ jessie/main armhf Packages
```

**Comment by ylevtov** [comment]
Hmm, I think I managed to install it. Thanks to your tip I looked into installing previous versions of `libola1` (`0.9.1-1.1` requires `libprotobuf9` while version `0.9.8-1` requires `libprotobuf7` which doesn't work on the RPi3) and `ola`.

To summarise for anyone else in the same position, to install `ola` on an RPi 3 via `apt-get` do:
```sudo apt-get install ola=0.9.1-1.1```

**Comment by ylevtov** [comment]
Next issue with this approach, however; It would seem that the plugins aren't in there right place somehow?
```
pi@raspberrypi:~ $ olad
IOUtils.cpp:38: Failed to open /dev/kldmx0: No such file or directory
KaratePlugin.cpp:79: Could not open /dev/kldmx0 No such file or directory
IOUtils.cpp:38: Failed to open /dev/dmx0: No such file or directory
OpenDmxPlugin.cpp:80: Could not open /dev/dmx0 No such file or directory
```
Is there something separate I have to do to install these plugins if installing from `apt-get`?

**Comment by peternewman** [comment]
You really want to remove/comment out the http://apt.openlighting.org/raspbian/ line that will be somewhere, probably in /etc/apt/sources.list or /etc/apt/sources.list.d and then it won't find and try and install software that's not designed for your OS.

Those plugins are all fine. You could disable them if you really wanted, otherwise they'll burst into life when you plug in the appropriate dongles.

**Comment by ylevtov** [comment]
Got it. I think I added that because I was trying all sorts of ways of installing it, including [this](http://opendmx.net/index.php/OLA_Debian_/_Ubuntu) one along the way, which says to add that link to /etc/apt/sources.list.

And thanks for clarifying regarding the plugins. This leads me to believe that my OpenDMX has bitten the dust, as it's not appearing in `/dev/` when it's plugged in. `dmx0` should appear in the /dev/ directory when simply plugged in via USB, right?

**Comment by peternewman** [comment]
Yeah some of those instructions have gone a bit stale since newer versions of Raspbian have come out.

Your OpenDMX is most likely fine, see the notes here, essentially you either need to enable the FTDI plugin, or install the kernel module:
https://www.openlighting.org/ola/get-help/ola-faq/#What_is_the_difference_between_the_different_USB_plugins

**Comment by ylevtov** [comment]
Great - will try that tonight. Many thanks!

**Comment by homeplayer2** [comment]
On this page http://docs.openhab.org/addons/bindings/dmx1/readme.html#installing-ola-binaries

It told us to ad the line to /etc/apt/sources.list 
comment out the line http://apt.openlighting.org/raspbian/ did fix the error for me

**Comment by tonys0** [comment]
Hi guys,
the image raspbian-ola-0.9.5.zip from http://dl.openlighting.org/ does not boot on Pi3, nor Pi2. The Pi stays in the rainbow screen with the undervoltage icon and does not go further. Nor is the ACT LED flashing.
I have tried to follow your sophisticated discussion here, but I do not think I would be able to fix the problem.
If the reason for giving out the image was to make it simple for the user, than it is completely wrong, making the user even more confused.
Please, fix the image or give the people a good installation instruction, starting from booting on a reliable Rasbian image.
Thank you very much.
